driver/src/test/functional/com/mongodb/client/CrudTest.java

```diff
@@ -0,0 +1,271 @@
+        }
+        return toResult(findIterable);
+    }
+
+    private BsonDocument getDeleteManyResult(final BsonDocument arguments) {
+        return toResult("deletedCount",
+                new BsonInt32((int) collection.deleteMany(arguments.getDocument("filter")).getDeletedCount()));
+    }
+
+    private BsonDocument getDeleteOneResult(final BsonDocument arguments) {
+        return toResult("deletedCount", new BsonInt32((int) collection.deleteOne(arguments.getDocument("filter")).getDeletedCount()));
+    }
+
+    private BsonDocument getFindOneAndDeleteResult(final BsonDocument arguments) {
+        FindOneAndDeleteOptions options = new FindOneAndDeleteOptions();
+        if (arguments.containsKey("projection")) {
+            options.projection(arguments.getDocument("projection"));
+        }
+        if (arguments.containsKey("sort")) {
+            options.sort(arguments.getDocument("sort"));
+        }
+        return toResult(collection.findOneAndDelete(arguments.getDocument("filter"), options));
+    }
+
+    private BsonDocument getFindOneAndReplaceResult(final BsonDocument arguments) {
+        FindOneAndReplaceOptions options = new FindOneAndReplaceOptions();
+        if (arguments.containsKey("projection")) {
+            options.projection(arguments.getDocument("projection"));
+        }
+        if (arguments.containsKey("sort")) {
+            options.sort(arguments.getDocument("sort"));
+        }
+        if (arguments.containsKey("upsert")) {
+            options.upsert(arguments.getBoolean("upsert").getValue());
+        }
+        if (arguments.containsKey("returnDocument")) {
+            options.returnOriginal(!arguments.getString("returnDocument").getValue().equals("After"));
+        }
+        return toResult(collection.findOneAndReplace(arguments.getDocument("filter"), arguments.getDocument("replacement"), options));
+    }
+
+    private BsonDocument getFindOneAndUpdateResult(final BsonDocument arguments) {
+        FindOneAndUpdateOptions options = new FindOneAndUpdateOptions();
+        if (arguments.containsKey("projection")) {
+            options.projection(arguments.getDocument("projection"));
+        }
+        if (arguments.containsKey("sort")) {
+            options.sort(arguments.getDocument("sort"));
+        }
+        if (arguments.containsKey("upsert")) {
+            options.upsert(arguments.getBoolean("upsert").getValue());
+        }
+        if (arguments.containsKey("returnDocument")) {
+            options.returnOriginal(!arguments.getString("returnDocument").getValue().equals("After"));
+        }
+        return toResult(collection.findOneAndUpdate(arguments.getDocument("filter"), arguments.getDocument("update"), options));
+    }
+
+    private BsonDocument getInsertOneResult(final BsonDocument arguments) {
+        collection.insertOne(arguments.getDocument("document"));
+        return toResult((BsonValue) null);
```

Comment:
It's a one line change in the test code, so the complexity argument doesn't hold for me.  It's going to be more trouble to keep modifying the json file by hand every time tests are added.